public/components/CampaignAnalytics/Analytics/CampaignTrafficDriverStatsChart.js

```diff
@@ -0,0 +1,172 @@
+import React, {PropTypes} from "react";
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+  ReferenceLine,
+  Brush
+} from "recharts";
+import {connect} from "react-redux";
+import {bindActionCreators} from "redux";
+import * as getCampaignTrafficDriverStats from "../../../actions/CampaignActions/getCampaignTrafficDriverStats";
+
+class CampaignTrafficDriverStatsChart extends React.Component {
+
+  componentWillMount() {
+    this.props.campaignTrafficDriverStatsActions.getCampaignTrafficDriverStats(this.props.campaign.id);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.campaign.id !== this.props.campaign.id) {
+      this.props.campaignTrafficDriverStatsActions.getCampaignTrafficDriverStats(this.props.campaign.id);
+    }
+  }
+
+  renderDriverGroupChart = (group) => {
+
+    const data = group.dayStats.map((day) => {
+      return (
+      {
+        date: day.date,
+        impressions: day.stats.impressions,
+        clicks: day.stats.clicks,
+        ctr: Math.round(day.stats.ctr * 100) / 100
+      }
+      );
+    });
+
+    const anchorName = group.groupName.toLowerCase().split(' ').join('-') + '-drivers';
+
+    const shortDateFormat = (date) => {
+      return new Date(date).toLocaleDateString('en-GB', {day: '2-digit', month: '2-digit'})
+    };
+
+    const longerDateFormat = (date) => {
+      return new Date(date).toLocaleDateString('en-GB', {weekday: 'short', day: '2-digit', month: 'short'})
+    };
```

Comment:
There is a date format helper js file that is used to process dates elsewhere in the js layer (see dateFormatter.js). Might be worth moving these date ops into that file or checking if the existing formatter functions do the job already